src/components/ProductCard.tsx

```diff
@@ -33,39 +28,36 @@ export default function ProductCard({
     }, 350);
   };
 
-  const changeIsCardFlipped = () => {
+  const toggleIsFlippedCard = () => {
     setIsCardFlipped(!isCardFlipped);
   };
 
+  const rotateCardImage = isCardFlipped
+    ? "-rotateY(180deg)"
+    : "rotateY(180deg)";
+
   return (
-    <div
-      key={key}
-      className=" group flex flex-col flex-wrap justify-center border-4 bg-gray-100 border-white"
-    >
+    <div className=" group flex flex-col flex-wrap justify-center border-4 bg-gray-100 border-white">
       <div className="flex flex-row relative justify-center">
         <ProductCardShowMoreButton
           isCardFlipped={isCardFlipped}
-          changeIsCardFlipped={changeIsCardFlipped}
+          toggleIsFlippedCard={toggleIsFlippedCard}
           changeImage={changeImage}
           content={content}
         />
       </div>
       <div className="flex flex-col w-[413.33px] h-[508px] items-center relative bg-gray-100">
         <div className="top-10 relative">
           <div
-            className={`${
-              isCardFlipped
-                ? "[transform:-rotateY(180deg)]"
-                : "[transform:rotateY(180deg)]"
-            } ease-in-out [transform-style-:preserve-3d] [backface-visblity:hidden] transition-all duration-1000`}
+            className={`[transform:${rotateCardImage}] ease-in-out [transform-style-:preserve-3d] [backface-visblity:hidden] transition-all duration-1000`}
           >
             <Image
-              className={`transition-all ease-in-out duration-300`}
+              className="transition-all ease-in-out duration-300"
               src={cardImage}
               height={300}
               width={300}
               alt="cardImage"
-              style={{ opacity: opacity / 1 }}
+              style={{ opacity: opacity }}
```

Comment:
this can be simplified to this. 
```suggestion
              style={{ opacity }}
```

Reply:
added.